plugin/storage/es/mappings/jaeger-dependencies-6.json

```diff
@@ -1,8 +1,8 @@
 {
   "template": "*jaeger-dependencies-*",
   "settings":{
-    "index.number_of_shards": {{ .Shards }},
-    "index.number_of_replicas": {{ .Replicas }},
+    "index.number_of_shards": {{ .Indices.Dependencies.TemplateNumShards }},
```

Comment:
I think .Indices.Dependencies should be taken care of by the caller to this template, no need to create this coupling between the template and an external overly broad config.

Reply:
I rolled back the template changes as you suggested, tried to differentiate between the magging calls, but wasn't sure if that was the best way to do it. I defined `IndexTemplateOptions` structure. `IndexTemplateOptions` is primarily used when `tmpl.Execute` is executed with only a specific index parameters. Because we can't do that with a `MappingBuilder`